analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SillyBitwiseOperation.cs

```diff
@@ -86,4 +86,65 @@ public void Repro(object[] args)
     }
 
     private bool CheckArg(object arg) => false;
+
+    public void FindConstant_For()
+    {
+        var value = 0;
+        var unchanged = 0;
+        int result;
+        for (var v = 0; v < 42; v++)
+        {
+            result = value | v;     // Compliant, value changes over iterations
```

Comment:
could you also add the same, but with `value = 1` at the beginning of the loop block?

```
for (...)
{
  value = 1;
  result = value | v; // Noncompliant
  result = unchanged | v; // Noncompliant
}
```

and also with 

```
for (...)
{
  result = value | v; // Noncompliant
  result = unchanged | v; // Noncompliant
  value = 0; // same value as initial one
}
```

(same for VB)

even if it won't work as expected, it's good to document the cases